### PR TITLE
Publish topic (move_base_simple/feedback) with path length and time left to goal

### PR DESCRIPTION
**Pull request by michaelblindnology**
Publish topic (move_base_simple/feedback) with path length and time left to goal

### CODE REVIEW

**Comment by pavloblindnology** [comment]
As @mikeferguson said:
> we should not be adding messages to move_base - there is a move_base_msgs package in a separate repo specifically so that users of nav do not have to build all of nav on remote machines that are monitoring outputs.
>
> There is also a standard path message in nav_msgs, why not use that?

`Path` msg type name is completely non-descriptive.
Let's also wait for maintainers to respond if it's ok to extend action feedback.

File: move_base/src/move_base.cpp [hunk 1]
```diff
@@ -1200,4 +1207,19 @@ namespace move_base {
...
+
+  // publish in context of action feedback, path left and time left to the goal
+  void MoveBase::publishPathMsg(const geometry_msgs::PoseStamped& current_position, const geometry_msgs::PoseStamped& goal)
+  {
+    move_base::Path msg;
+    msg.current_pose = current_position;
+    msg.goal_pose = goal;
+    msg.dist_to_goal = distance(current_position, goal);
+    geometry_msgs::Twist cmd_vel;
+    tc_->computeVelocityCommands(cmd_vel);
```
Comment (pavloblindnology) [inline move_base/src/move_base.cpp:1219]: This function is quite heavy and result is veeery variable, use max speed.

File: move_base/src/move_base.cpp [hunk 2]
```diff
@@ -1200,4 +1207,19 @@ namespace move_base {
...
     return true;
   }
+
+  // publish in context of action feedback, path left and time left to the goal
+  void MoveBase::publishPathMsg(const geometry_msgs::PoseStamped& current_position, const geometry_msgs::PoseStamped& goal)
+  {
+    move_base::Path msg;
+    msg.current_pose = current_position;
+    msg.goal_pose = goal;
+    msg.dist_to_goal = distance(current_position, goal);
```
Comment (pavloblindnology) [inline move_base/src/move_base.cpp:1217]: Currently, you compute straight line distance instead of global plan length left.

**Review by pavloblindnology — CHANGES_REQUESTED** [review]

File: move_base/src/move_base.cpp (hunk 2 above)
Comment (michaelblindnology) [reply move_base/src/move_base.cpp:1217]: I know, but I can not call planner in the context of action

Comment (pavloblindnology) [reply move_base/src/move_base.cpp:1217]: You don't need to compute it, it's already precomputed (look for `latest_plan_`, `controller_plan_`)

File: move_base/src/move_base.cpp (hunk 1 above)
Comment (michaelblindnology) [reply move_base/src/move_base.cpp:1219]: Ok. What is the class member of max_speed?

File: move_base/src/move_base.cpp (hunk 2 above)
Comment (michaelblindnology) [reply move_base/src/move_base.cpp:1217]: Thanks

File: move_base/src/move_base.cpp (hunk 1 above)
Comment (pavloblindnology) [reply move_base/src/move_base.cpp:1219]: Have no idea. It's actually loaded by local planner and not exposed to move_base. But you can load it from rosparam and dynamical reconfigure.

Comment (michaelblindnology) [reply move_base/src/move_base.cpp:1219]: Thanks a lot.

Comment (pavloblindnology) [reply move_base/src/move_base.cpp:1219]: Or expose it to move_base somehow.

**Pull request closed without merging**